plugins/frontend/backstage-plugin-iframe/src/components/IFrameComponent.tsx

```diff
@@ -16,41 +16,118 @@
 
 import React from 'react';
 import { ErrorComponent } from './ErrorComponent';
-import { IFrameProps } from './types';
+import {
+  IFrameProps,
+  IFrameComponentProps,
+  IFrameFromAnnotationProps,
+} from './types';
 import { Content, ContentHeader } from '@backstage/core-components';
 import { configApiRef, useApi } from '@backstage/core-plugin-api';
 import { determineError } from './utils/helpers';
+import { useEntity } from '@backstage/plugin-catalog-react';
 
-/**
- * A component to render a IFrame
- *
- * @public
- */
-export const IFrameCard = (props: IFrameProps) => {
-  const { src, height, width } = props;
+type IFrameComponentContentProps = {
+  src: string;
+  title: string;
+  height?: string;
+  width?: string;
+};
+
+const IFrameComponentContent = (props: IFrameComponentContentProps) => {
+  const { title, src, height, width } = props;
+
+  return (
+    <Content>
+      <ContentHeader title={title} />
+      <iframe
+        src={src}
+        height={height || '100%'}
+        width={width || '100%'}
+        title={title}
+      />
+    </Content>
+  );
+};
+
+const IFrameFromAnnotation = (props: IFrameFromAnnotationProps) => {
+  const { srcFromAnnotation, height, width } = props;
   const title =
     props.title || 'Backstage IFrame (Note you can modify this with the props)';
   const configApi = useApi(configApiRef);
   const allowList = configApi.getOptionalStringArray('iframe.allowList');
-  const errorMessage = determineError(src, allowList);
+  const { entity } = useEntity();
+  let errorMessage = '';
 
-  if (errorMessage !== '') {
-    return <ErrorComponent {...{ errorMessage }} />;
+  const src = entity.metadata.annotations?.[srcFromAnnotation];
+
+  if (!src) {
+    return (
+      <ErrorComponent
+        errorMessage={`Failed to get url src from the entity annotation ${srcFromAnnotation}`}
+      />
+    );
+  }
+
+  errorMessage = determineError(src, allowList);
```

Comment:
I know I'm being a bit annoying about this but we should probably check the src url protocol is not `javascript:` (or that it is http/https) while we're validating (we would get that indirectly if the user sets an allow list but out of the box it's insecure)

Reply:
I did something similar, except I am insisting it must be a https url. Can you think of a reason why we might need to support other protocols for the iframe plugin?